packages/http/src/validator/validators/body.ts

```diff
@@ -86,12 +88,34 @@ function deserializeAndValidate(content: IMediaTypeContent, schema: JSONSchema,
   );
 }
 
+/**
+ * Given a schema, return an equivalent schema that does not include `allOf`.
+ */
+function withoutAllOf(s: JSONSchema): JSONSchema {
+  try {
+    return mergeAllOf(s, {
+      // `deep` traverses the *whole* schema.  If this becomes a performance
+      // problem, see the discussion and alternate implementation at
+      // https://github.com/stoplightio/prism/pull/1957/files#r760950082
```

Comment:
After discussing this with @P0lip further, we've decided to leave the `deep` allOf merging option enabled for now.  I've added a note in the code that points to this discussion should poor performance become a problem later.